doc/adr/0009-kryo-class-registration-policy.md

```diff
@@ -0,0 +1,40 @@
+# 9. Kryo class registration policy
+
+Date: 2022-02-17
+
+## Status
+
+Proposed
+
+## Context
+
+Having reliable serialization across different modules require using consistent and 
+non-conflicting registration ids.
+
+## Decision
+
+* Each module containing classes that are a subject of a kryo serialization declares 
+its kryo registrar - a public constant of type `Map[Class[_], Int]` representing kryo 
```

Comment:
As a bonus feature, we can check if `Class[_]` have duplicates and if `Int` have duplicates and throw if it happens as we talked.

Reply:
Fully agree, that's an implementation detail however I think so I would like to not include it in ADR.